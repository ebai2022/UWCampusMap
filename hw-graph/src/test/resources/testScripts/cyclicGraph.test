# Test a cyclic graph

# Create a graph
CreateGraph A

# Add nodes
AddNode A e1
AddNode A e2

# Add edges
AddEdge A e1 e2 forwards
AddEdge A e2 e1 backwards

# Check if the graph is cyclic
ListChildren A e1
ListChildren A e2