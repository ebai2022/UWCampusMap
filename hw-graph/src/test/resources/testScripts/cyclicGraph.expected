# Test a cyclic graph

# Create a graph
created graph A

# Add nodes
added node e1 to A
added node e2 to A

# Add edges
added edge forwards from e1 to e2 in A
added edge backwards from e2 to e1 in A

# Check if the graph is cyclic
the children of e1 in A are: e2(forwards)
the children of e2 in A are: e1(backwards)